code/game/objects/items/weapons/holy_weapons.dm

```diff
@@ -301,6 +305,67 @@
 		qdel(S)
 	return ..()
 
+/obj/item/nullrod/scythe/talking/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/soulstone) || !possessed)
+		return ..()
+	if(obj_integrity >= max_integrity)
+		to_chat(user, "<span class='notice'>You have no reason to replace a perfectly good soulstone with a new one.</span>")
+		return
+	to_chat(user, "<span class='notice'>You load a new soulstone into the possessed blade.</span>")
+	playsound(user, 'sound/weapons/gun_interactions/shotgunpump.ogg', 60, TRUE)
+	obj_integrity = max_integrity
+	for(var/mob/living/simple_animal/shade/sword/sword_shade in contents)
+		sword_shade.health = sword_shade.maxHealth
+	qdel(I)
+
+/obj/item/nullrod/scythe/talking/take_damage(damage_amount)
+	if(possessed)
+		for(var/mob/living/simple_animal/shade/sword/sword_shade in contents)
+			sword_shade.take_overall_damage(damage_amount)
+	return ..()
+
+/obj/item/nullrod/scythe/talking/proc/click_actions(atom/attacking_atom, mob/living/simple_animal/attacking_shade)
+	if(world.time <= attacking_shade.next_move) // yea we gotta check
+		return
+	if(!ismovable(attacking_atom))
+		return
+	attacking_shade.changeNext_move(CLICK_CD_MELEE)
+	if(ismob(loc))
+		var/mob/living/carbon/human/our_location = loc
+		if(istype(our_location))
+			if(src != our_location.l_hand && src != our_location.r_hand)
+				return
+			if(our_location.Adjacent(attacking_atom)) // with a buddy we deal 10 damage :D
+				our_location.do_attack_animation(attacking_atom, used_item = src)
+				melee_attack_chain(attacking_shade, attacking_atom)
+			return
+	if(Adjacent(attacking_atom)) // without a buddy we only deal 5 damage :c
```

Comment:
early return if the `get_turf(src)` isn't adjancent to `attacking_atom` rather than checking it twice, here and on line 338.

Reply:
won't that just make it so that if you are in an object with the sword you can attack through the object? Like a mecha for example

Reply:
Shouldn't be any logic change, just an early return.

Reply:
Discussed on discord a bit. The concern is that calling `Adjacent` on a turf would make you able to attack through objects, However `atom/movable/Adjacent` (which is called from this parent) also calls `Adjacent` on a turf, so I'm not there will be any weird behavior here.

Reply:
This breaks shit